airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/datetime_stream_state_converter.py

```diff
@@ -183,7 +183,7 @@ def __init__(
         self._parser = DatetimeParser()
 
     def output_format(self, timestamp: datetime) -> str:
-        return timestamp.strftime(self._datetime_format)
+        return self._parser.format(timestamp, self._datetime_format)
```

Comment:
Similar to https://github.com/airbytehq/airbyte/pull/48361, although it adds a little bit of a dependency on the declarative framework, it's safer to retain the exact behavior by using the parser. And we can always move this code into concurrent cursor later